### DISCUSSION TITLE
Unable to install Windows App SDK C# VS2022 Templates in latest VS2022

### DISCUSSION THREAD

**Discussion by jsymon** [microsoft/WindowsAppSDK#2014] [Q&A] [answered]
Attempting to install the Windows App SDK .vsix on the latest Visual Studio 2022 (Version 17.1.0 Preview 3.0)

![image](https://user-images.githubusercontent.com/6398627/150957176-0c61b502-4cb1-46c6-940a-5c64bf582c9f.png)

Presents the error:

Installation of Microsoft.WindowsAppSDK.Cs.Dev17 failed. The extension has a lower version than required by Visual Studio.  Please install the extension from Visual Studio Installer instead.

Identifier         : Microsoft.WindowsAppSDK.Cs.Dev17
Name               : Windows App SDK C# VS2022 Templates
Author             : Microsoft
Version            : 1.0.0.59133071

However, I'm pretty sure the latest version that can be selected from the 'online' extension explorer inside Visual Studio, is the 1.0 experimental (and therefore older than this release).

**Comment by adamplonka** [comment]
Same for Version 17.1.0 Preview 4.0
last version it worked on was preview 2.0

**Comment by BenJKuhn** [comment, marked as answer]
I believe this is because on Preview 3 or higher, the App SDK is available through VS installer, rather than marketplace. @DefaultRyan, can you confirm?

Ben

**Reply by DefaultRyan** [reply]
Correct. As of Windows App SDK 1.0 Preview 3 and VS 2022 17.1 Preview 1, you get the Windows App SDK extension from the Visual Studio installer, not from the marketplace or download center. See the attached screenshot.

<img width="960" alt="image" src="https://user-images.githubusercontent.com/26174284/151447143-daa41cad-17ec-4f07-9799-873111f8b8c5.png">


We'll get the guidance at https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/set-up-your-development-environment?tabs=vs-2022 updated to reflect this.

**Reply by adamplonka** [reply]
@DefaultRyan: This is really badly translated in polish version:
"Szablony języka C# zestawu SDK do aplikacji systemu Windows" / "C# language templates of SDK set for Windows system applications".
No "Windows App SDK" mentioned anywhere. I would never guess that's it without your screenshot

**Reply by DefaultRyan** [reply]
Thanks for reporting that @plonkaadam . I'll work with localization to get the translation quality improved.

**Reply by jeromelaban** [reply]
Also FYI, the search tool in individual components is not showing the templates:

<img width="969" alt="image" src="https://user-images.githubusercontent.com/5839577/152875847-80d21a5b-685e-4478-94da-dc51a0cfc4b9.png">

It's available in the .NET desktop development section:

![image](https://user-images.githubusercontent.com/5839577/152875945-a81f5ad1-4bce-40f7-a483-d29a9389facb.png)

**Reply by BenJKuhn** [reply]
@jeromelaban, that's a great catch. That will require an update in the Visual Studio project, though. We don't control that from the App SDK content side of things. Would you be able to file that as an issue in Visual Studio using the "send feedback" button in VS in the upper right, beneath the close icon? If you'd prefer, I can do it, but it was your discovery, so you get first right of refusal.

Ben

**Reply by jeromelaban** [reply]
I will open an feedback, thanks!

**Reply by jeromelaban** [reply]
Note, this is still not fixed as of VS 2022 17.3 Preview 2.

**Reply by DefaultRyan** [reply]
Do you have a link to the feedback item you opened? Is it still opened?

**Reply by jeromelaban** [reply]
It was closed as not a bug: https://developercommunity.visualstudio.com/t/Searching-for-Windows-App-SDK-C-Templa/1663318?space=8&q=Windows+App+SDK+template
